in_toto/model.go

```diff
@@ -171,6 +317,32 @@ type Metablock struct {
 	Signatures []Signature `json:"signatures"`
 }
 
+func checkRequiredFields(signed map[string]interface{}) error {
+	reflection := reflect.TypeOf(nil)
+	if signed["_type"] == "link" {
+		var link Link
+		reflection = reflect.TypeOf(link)
+	} else if signed["_type"] == "layout" {
+		var layout Layout
+		reflection = reflect.TypeOf(layout)
+	} else {
+		return fmt.Errorf("The '_type' field of the 'signed' part of in-toto" +
+			" metadata must be one of 'link' or 'layout'")
+	}
```

Comment:
I suggest to call `reflect.TypeOf` in `Load` below, where you already have`Link`/`Layout` switch, and pass it as an additional argument to `checkRequiredFields` inside the switch. What do you think?

Reply:
Unfortunately, I'm unable to pass a value of type `reflect.rtype` because it's taken to be private? Seems like it's the case because it's not capitalized in the `reflect` package.

Reply:
Why not pass `reflect.Type`...?
```golang
func checkRequiredFields(signed map[string]interface{}, reflection reflect.Type) error {
  attributeCount := reflection.NumField()
  //...
```

And you call in `Load` below:
```golang
// ...
if signed["_type"] == "link" {
  var link Link
  if err := checkRequiredFields(signed, reflect.TypeOf(link)); err != nil {
    return err
  }
  // ...  repeat for `signed["_type"] == "layout" `
```

Reply:
I didn't realize .Type was the same thing, apologies.

Reply:
No worries. ;)